crmsh/bootstrap.py

```diff
@@ -105,6 +106,225 @@ def init_qdevice(self):
                 cmds=self.qdevice_heuristics,
                 mode=self.qdevice_heuristics_mode)
 
+    def init_sbd_manager(self):
+        self.sbd_manager = SBDManager(self.sbd_devices, self.diskless_sbd)
+
+
+class SBDManager(object):
+    """
+    Class to manage sbd configuration and services
+    """
+    SYSCONFIG_SBD_TEMPLATE = "/usr/share/fillup-templates/sysconfig.sbd"
+    SBD_STATUS_DESCRIPTION = """
+Configure SBD:
+  If you have shared storage, for example a SAN or iSCSI target,
+  you can use it avoid split-brain scenarios by configuring SBD.
+  This requires a 1 MB partition, accessible to all nodes in the
+  cluster.  The device path must be persistent and consistent
+  across all nodes in the cluster, so /dev/disk/by-id/* devices
+  are a good choice.  Note that all data on the partition you
+  specify here will be destroyed.
+"""
+
+    def __init__(self, sbd_devices=None, diskless_sbd=False):
+        """
+        Init function
+
+        sbd_devices is provided by '-s' option on init process
+        diskless_sbd is provided by '-S' option on init process
+        """
+        self.sbd_devices_input = sbd_devices
+        self.diskless_sbd = diskless_sbd
+        self._sbd_service_flag = False
+        self._sbd_devices = None
+
+    @staticmethod
+    def _check_environment():
+        """
+        Check prerequisites for SBD
+        """
+        if not check_watchdog():
+            error("Watchdog device must be configured in order to use SBD")
+        if not utils.is_program("sbd"):
+            error("sbd executable not found! Cannot configure SBD")
+
+    def _parse_sbd_device(self):
+        """
+        Parse sbd devices, possible command line is like:
+          -s "/dev/sdb1;/dev/sdb2"
+          -s /dev/sdb1 -s /dev/sbd2
+        """
+        result_list = []
+        for dev in self.sbd_devices_input:
+            if ';' in dev:
+                result_list.extend(dev.strip(';').split(';'))
+            else:
+                result_list.append(dev)
+        return result_list
+
+    @staticmethod
+    def _verify_sbd_device(dev_list):
+        """
+        Verify sbd device
+        """
+        if len(dev_list) > 3:
+            raise ValueError("Maximum number of SBD device is 3")
+        for dev in dev_list:
+            if not is_block_device(dev):
+                raise ValueError("{} doesn't look like a block device".format(dev))
+
+    def _get_sbd_device_interactive(self):
+        """
+        Get sbd device on interactive mode
+        """
+        if _context.yes_to_all:
+            warn("Not configuring SBD (%s left untouched)." % (SYSCONFIG_SBD))
+            return
+
+        status(self.SBD_STATUS_DESCRIPTION)
+
+        if not confirm("Do you wish to use SBD?"):
+            warn("Not configuring SBD - STONITH will be disabled.")
```

Comment:
I don't think we are going to turn off `stonith-enabled` cluster property in this case, right? To avoid confusions, we could  probably say something like  "Not configuring SBD - Fencing won't work without fencing resources being configured ".